snapcraft/tests/test_plugin_catkin.py

```diff
@@ -447,19 +447,50 @@ def test_prepare_build(self):
     @mock.patch.object(catkin.CatkinPlugin, 'run_output', return_value='foo')
     def test_finish_build(self, run_output_mock, run_mock):
         plugin = catkin.CatkinPlugin('test-part', self.properties)
-        os.makedirs(plugin.rosdir)
+        os.makedirs(os.path.join(plugin.rosdir, 'bin'))
+        os.makedirs(os.path.join(plugin.rosdir, 'etc', 'catkin', 'profile.d'))
+
+        shebang_file1 = os.path.join(plugin.rosdir, '_setup_util.py')
+        shebang_file2 = os.path.join(plugin.rosdir, 'bin', 'catkin_find')
 
-        # Place _setup_util.py with a bad shebang
-        with open(os.path.join(plugin.rosdir, '_setup_util.py'), 'w') as f:
+        # Place a few files with bad shebangs
+        with open(shebang_file1, 'w') as f:
             f.write('#!/foo/bar/baz/python')
+        with open(shebang_file2, 'w') as f:
+            f.write('#!/foo/baz/python')
+
+        # Place a file with no shebang
+        irrelevant_file = os.path.join(plugin.rosdir, 'foo')
+        with open(irrelevant_file, 'w') as f:
+            f.write('foo')
+
+        ros_profile = os.path.join(plugin.rosdir, 'etc', 'catkin', 'profile.d',
+                                   '10.ros.sh')
+
+        # Place 10.ros.sh with an absolute path to python
+        with open(ros_profile, 'w') as f:
+            f.write('/usr/bin/python foo')
 
         plugin._finish_build()
 
-        # Verify that _setup_util.py had its shebang rewritten correctly
-        with open('{}/_setup_util.py'.format(
-                plugin.rosdir), 'r') as f:
-            self.assertEqual(f.read(), '#!/usr/bin/env python',
-                             'The shebang was not replaced as expected')
+        # Verify that the shebangs were rewritten correctly
+        for shebang_file in [shebang_file1, shebang_file2]:
+            with open(shebang_file, 'r') as f:
+                self.assertEqual(f.read(), '#!/usr/bin/env python',
+                                 'The shebang in "{}" was not replaced as '
+                                 'expected'.format(shebang_file))
+
+        # Verify that the irrelevant file wasn't touched
+        with open(irrelevant_file, 'r') as f:
+            self.assertEqual(f.read(), 'foo',
+                             'Expected the contents of "{}" to remain '
+                             'unchanged.'.format(irrelevant_file))
+
+        # Verify that the absolute path in 10.ros.sh was rewritten correctly
+        with open(ros_profile, 'r') as f:
+            self.assertEqual(f.read(), 'python foo',
+                             'The absolute path to python was not replaced as '
+                             'expected')
```

Comment:
I instead refactored the test into two, one testing the shebang rewrite and the other testing the absolute path rewrite. It can be refactored further at some point if we get rid of _finish/_prepare_build(), but that's for a different PR.